openupgrade_scripts/scripts/mail/15.0.1.5/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,266 @@
+# DONE: post-migration, mapped old field values force_next (true -> trigger, false -> suggest)
+
+mail         / mail.activity.type       / default_note (html)           : NEW
+mail         / mail.activity.type       / default_description (html)    : DEL
+# DONE: pre-migration, rename fields default_description -> default_note
+
+mail         / mail.activity.type       / res_model (selection)         : NEW selection_keys: function
+mail         / mail.activity.type       / res_model_id (many2one)       : DEL relation: ir.model
+# DONE: post-migration: mapped values res_model_id to res_model
+
+mail         / mail.activity.type       / next_type_ids (many2many)     : DEL relation: mail.activity.type
+mail         / mail.activity.type       / default_next_type_id (many2one): DEL relation: mail.activity.type
+mail         / mail.activity.type       / suggested_next_type_ids (many2many): NEW relation: mail.activity.type, isfunction: function, stored
+mail         / mail.activity.type       / triggered_next_type_id (many2one): NEW relation: mail.activity.type, isfunction: function, stored
+# DONE: pre-migration, rename fields (default_next_type_id -> triggered_next_type_id, next_type_ids -> suggested_next_type_ids)
+
+mail         / mail.channel             / channel_message_ids (many2many): DEL relation: mail.message
+# NOTHING TO DO: feature removed
+
+mail         / mail.channel             / channel_partner_ids (many2many): not stored anymore
+mail         / mail.channel             / channel_partner_ids (many2many): now a function
+mail         / mail.channel             / channel_partner_ids (many2many): table is now 'False' ('mail_channel_partner')
+# NOTHING TO DO: table will be removed by ORM
+
+mail         / mail.channel             / channel_type (selection)      : selection_keys is now '['channel', 'chat', 'group']' ('['channel', 'chat']')
+# NOTHING TO DO: new feature
+
+mail         / mail.channel             / rtc_session_ids (one2many)    : NEW relation: mail.channel.rtc.session
+mail         / mail.channel             / default_display_mode (selection): NEW selection_keys: ['video_full_screen']
+# NOTHING TO DO: new feature call audio and video
+
+mail         / mail.channel             / email_send (boolean)          : DEL
+mail         / mail.channel             / moderation (boolean)          : DEL
+mail         / mail.channel             / moderation_guidelines (boolean): DEL
+mail         / mail.channel             / moderation_guidelines_msg (text): DEL
+mail         / mail.channel             / moderation_ids (one2many)     : DEL relation: mail.moderation
+mail         / mail.channel             / moderation_notify (boolean)   : DEL
+mail         / mail.channel             / moderation_notify_msg (text)  : DEL
+mail         / mail.channel             / moderator_ids (many2many)     : DEL relation: res.users
+# NOTHING TO DO: new feature
```

Comment:
Is it the case that channel functionality was split off to the new module mail_group? Do we need to prepare for it by renaming the models and fields in this migration, and select 'mail_group' for installation if there is actual data in these tables (or apply some other strategy to ensure the module is installed)?

Reply:
@StefanRijnhart yes, channel functionality was split off to the new module mail_group, but we don't have to do.
Simply keep the old mail.channel, and the old mail.channel no longer has the ability to send mail
`mail_group` is used for mass mailing, and mail.group will use for new data